### ISSUE TITLE
Cannot get multi-level nested properties using Delta.TryGetPropertyValue

### ISSUE THREAD

**Issue by thisjgu** [OData/WebApi#2260]
When reading a property with nested complex typed properties using Delta.TryGetPropertyValue(), the second level complex typed property is always null

### Assemblies affected
7.1.0.21120

### Reproduce steps
We are having such entityType
```
EntityTypeA:
{
     CompleTypeB compTypeB
     //other properties
}

ComplexTypeB:
{
     ComplexTypeC compTypeC
     ...//other properties
}
ComplexTypeC:
{
     string str
     ...//other properties
}
```
With `PATCH entityTypeA/entityTypeA_ID`:
```
{
    "compTypeB": {
        //other properties
        "compTypeC": {
            "str":"foobar"
        }
    }
}
```
### Expected result
Patched `EntityTypeA` should have
```
{
    "compTypeB": {
        //other properties
        "compTypeC": {
            "str":"foobar"
        }
    }
}
```

### Actual result
```
{
    "compTypeB": {
        //other properties
        "compTypeC": null
    }
}
```

### Additional detail
Looking into TryGetPropertyValue, it uses `_instance` value https://github.com/OData/WebApi/blob/master/src/Microsoft.AspNet.OData.Shared/DeltaOfTStructuralType.cs#L183

However during debugging, `_instance` doesn't appear to contain the value for nested properties. The nested property value only presents in `_deltaNestedResources`[propertyName].`_instance`

[This thread](https://github.com/OData/WebApi/issues/2238) mentioned the `multi level patch is not supported by design`. Is this true? If so is there any pointer to read about it. Also what's the best way to address this, if we don't want to expose a new API to patch the second level property?

**Comment by habbes** [comment]
Hi @thisjgu, this is actually a bug related to the issue https://github.com/OData/WebApi/issues/2241 and it was fixed by this PR: https://github.com/OData/WebApi/pull/2246

This fix will be included in the next release.

Meanwhile you can use `delta.Patch(original)` or `delta.CopyChangedValues(original)` to patch the nested properties to the original instance. Let me know if this helps.

**Comment by xuzhg** [comment]
@thisjgu Since the fix got merged, can you try our nightly bit? You can find the nightly information on the [readme.md](https://github.com/OData/WebApi/blob/master/README.md).

**Comment by thisjgu** [comment]
@xuzhg thanks for your reply. Due to our use case, we prefer to use prefer version. Can I know what's the schedule of next release?

Also for `delta.Patch(original)` or `delta.CopyChangedValues(original)`, my understanding is this only works if the logic knows the TStructuralType and has the underline entity, is that correct? 
(our use case talks to IDelta but not Delta<TStructural> directly, so trying to understand the feasibility)

**Comment by thisjgu** [comment]
@Sreejithpin @habbes could you please comment on the expected timeline for the next release?

**Comment by jananiva** [comment]
Gentle ping @Sreejithpin @habbes @xuzhg - could you provide us with the timelines for the next release?

**Comment by xuzhg** [comment]
@jananiva We are focusing on the next release of ODL.  So, it's a delay for the Web API,  but, we have a plan to release a new version. I hope it can finish within second half of next month.  If you are urgent, Maybe we can discuss it ahead to have a hotfix release for you to try.

**Comment by habbes** [comment]
@thisjgu could you share relevant snippets for your `Patch()` logic? You can use `IDelta<T>` to patch the original, e.g.

```c#
IActionResult Patch(int key, IDelta<Customer> delta)
{
    var originalCustomer = ...;
    delta.Patch(originalCustomer);
}
```

Maybe you can provide clarifications regarding your use case to see whether this can help.

**Comment by Kanac** [comment]
> Hi @thisjgu, this is actually a bug related to the issue #2241 and it was fixed by this PR: #2246
> 
> This fix will be included in the next release.
> 
> Meanwhile you can use `delta.Patch(original)` or `delta.CopyChangedValues(original)` to patch the nested properties to the original instance. Let me know if this helps.

Is the fix released yet? In which version if so?

**Comment by zdzislaw** [comment]
I solved it by duplicating the class:

Microsoft.AspNetCore.OData.Deltas namespace
public class Delta<T>


I needed to change the TrySetPropertyValuepublic method.

```
private bool TrySetPropertyValuepublic(string name, object value)
        {
            Debug.Assert(name != null, "Argument name is null");

            if (!(_allProperties.ContainsKey(name) && _updatableProperties.Contains(name)))
            {
                return false;
            }

            PropertyAccessor<T> cacheHit = _allProperties[name];

            if (value == null && !cacheHit.Property.PropertyType.IsNullable())
            {
                return false;
            }

            Type propertyType = cacheHit.Property.PropertyType;
            // Adjusting delta
            if (value != null && !TypeHelper.IsCollection(propertyType) &&
                (!propertyType.IsPrimitive &&
                !propertyType.IsAssignableFrom(value.GetType())) &&
                (propertyType.IsPrimitive && !CanConvert(value, propertyType)))
            {
                return false;
            }


            if (propertyType.IsPrimitive)
            {
                cacheHit.SetValue(_instance, Convert.ChangeType(value, propertyType));
            }
            else
            {
                cacheHit.SetValue(_instance, value);
            }


            _changedProperties.Add(name);
            return true;
        }

        public bool CanConvert(object? value, Type conversionType)
        {
            try
            {
                Convert.ChangeType(value, conversionType);
                return true;
            }
            catch (Exception)
            {
                return false;
            }
        }
```




I'll try to run some tests and see how I can commit to the official repository.

**Comment by KenitoInc** [comment]
@Kanac  This was released in v7.5.0 https://docs.microsoft.com/en-us/odata/changelog/webapi-7x#webapi-750
